Tensile/TensileCreateLib/KernelFileContext.py

```diff
@@ -0,0 +1,156 @@
+
+    Args:
+        outputPath: The directory path where the files should be opened.
+
+    Returns:
+        A tuple containing two `TextIOWrapper` objects for the source and header files respectively.
+
+    Notes:
+        - The source file is named "Kernels.cpp" and the header file is named "Kernels.h".
+        - Both files are opened in append mode with UTF-8 encoding.
+    """
+    srcFilename = Path(outputPath).resolve() / "Kernels.cpp"
+    hdrFilename = Path(outputPath).resolve() / "Kernels.h"
+    return open(srcFilename, "a", encoding="utf-8"), open(hdrFilename, "a", encoding="utf-8")
+
+
+def _openFilesBasedOnFirstKernel(kernelFiles: List[str]) -> Tuple[TextIOWrapper, TextIOWrapper]:
+    """Opens two files for appending based on the name of the first kernel file in the list.
+
+    Args:
+        kernelFiles: A list of kernel file names, where the first element is used to determine the base file name.
+
+    Returns:
+        A tuple containing two `TextIOWrapper` objects for appending to the source and header files respectively.
+
+    Notes:
+        - The base file name is derived by removing the ".cpp" extension from the first kernel file name.
+        - The source and header files are named using the base name with ".cpp" and ".h" extensions respectively.
+        - Files are opened in append mode with UTF-8 encoding.
+    """
+    filename = kernelFiles[0].replace(".cpp", "")
+    return open(filename + ".cpp", "a", encoding="utf-8"), open(
+        filename + ".h", "a", encoding="utf-8"
+    )
+
+
+class KernelFileContextManager:
+    """
+    A context manager for opening kernel files and ensuring they are closed after use.
+
+    This context manager uses the provided parameters to open kernel source and header files
+    and automatically closes them when exiting the context. It simplifies the management of
+    file resources in operations involving kernel file manipulation.
+
+    Usage:
+        with KernelFileContextManager(params, outputPath, kernelFiles) as (kernelSourceFile, kernelHeaderFile):
+            # Use kernelSourceFile and kernelHeaderFile here
+    """
+
+    def __init__(self, params: dict, outputPath: Path, kernelFiles: Optional[List[str]] = None):
+        """
+        Initializes the context manager with the necessary parameters for opening kernel files.
+
+        Args:
+            params (dict): A dictionary containing parameters for opening kernel files.
+            outputPath (Path): The output path where kernel files are located or will be created.
+            kernelFiles (Optional[List[str]]): A list of kernel file names.
+        """
+        self.numMergedFiles = params["NumMergedFiles"]
+        self.mergeFiles = params["MergeFiles"]
+        self.lazyLoading = params["LazyLibraryLoading"]
+        self.outputPath = outputPath
+        self.kernelFiles = kernelFiles
+        self.kernelSourceFile = None
+        self.kernelHeaderFile = None
```

Comment:
Are there any set of parameter values that we know are violating assumptions about this context manager that we can test here?

For example are all of the following ok:

1. mergeFiles = True, lazyLoading = True
2. mergeFiles = True, lazyLoading = False
3. mergeFiles = False, lazyLoading = True
4. mergeFiles = False, lazyLoading = False
5. All of the above with different values for numMergedFiles?
6. What if outputPath is empty?
7. What if kernelFiles is empty?

Also, rather than taking a dict should we take three params explicitly and type them?

Reply:
I like this. I can add in a method to verify pre-conditions and update the function to accept explicit arguments.